app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,175 @@
+            dialog.show()
+        } catch (e: IllegalStateException) {
+            Timber.e(e, "Could not show dialog.")
+        }
+    }
+
+    @JvmStatic fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable,
+        onNegativeBtnClick: Runnable
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String,
+        negativeButtonText: String,
+        onPositiveBtnClick: Runnable,
+        onNegativeBtnClick: Runnable
+    ) {
+        val builder = AlertDialog.Builder(activity)
+        builder.setTitle(title)
+        builder.setMessage(message)
+        if (!StringUtils.isBlank(positiveButtonText)) {
+            builder.setPositiveButton(
+                positiveButtonText
+            ) { dialogInterface: DialogInterface, _: Int ->
+                dialogInterface.dismiss()
+                onPositiveBtnClick.run()
+            }
+        }
+        if (!StringUtils.isBlank(negativeButtonText)) {
+            builder.setNegativeButton(
+                negativeButtonText
+            ) { dialogInterface: DialogInterface, _: Int ->
+                dialogInterface.dismiss()
+                onNegativeBtnClick.run()
+            }
+        }
+        val dialog = builder.create()
+        showSafely(activity, dialog)
+    }
+
+    /*
+    Shows alert dialog with custom view
+    */
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity, title: String, message: String, onPositiveBtnClick: Runnable,
+        onNegativeBtnClick: Runnable,
+        customView: View,
+        cancelable: Boolean
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick,
+            customView,
+            cancelable
+        )
+    }
+
+    /*
+    Shows alert dialog with custom view
+     */
+    private fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String,
+        negativeButtonText: String,
+        onPositiveBtnClick: Runnable,
+        onNegativeBtnClick: Runnable,
+        customView: View,
+        cancelable: Boolean
+    ) {
+        // If the custom view already has a parent, there is already a dialog showing with the view
+        // This happens for on resume - return to avoid creating a second dialog - the first one
+        // will still show
+
+        val builder = AlertDialog.Builder(activity)
```

Comment:
this is not really using the Builder pattern correctly, a Builder returns itself for every method call so you can chain them eg

```
AlertDialog.Builder(activity)
.setTitle(title)
.setMessage(message)
.create()
```

I also regularly favor using `apply` so I can omit the `.` calls

Reply:
> this is not really using the Builder pattern correctly, a Builder returns itself for every method call so you can chain them eg
> 
> ```
> AlertDialog.Builder(activity)
> .setTitle(title)
> .setMessage(message)
> .create()
> ```
> 

If I understood correctly, this should be the correct way?
```
val builder = AlertDialog.Builder(activity).apply {
            setTitle(title)
            setMessage(message)
            setView(customView)
            setCancelable(cancelable)
            setPositiveButton(positiveButtonText) { dialogInterface: DialogInterface, _: Int ->
                dialogInterface.dismiss()
                onPositiveBtnClick?.run()
            }
            setNegativeButton(negativeButtonText) { dialogInterface: DialogInterface, _: Int ->
                dialogInterface.dismiss()
                onNegativeBtnClick?.run()
            }
        }
        showSafely(activity, builder.create())
```

Reply:
That is how I want it to look, yes. I might even omit the builder variable altogether, just calling create after the apply block is closed

Reply:
Done!